libs/server/Storage/Session/ObjectStore/Common.cs

```diff
@@ -30,7 +30,7 @@ unsafe GarnetStatus RMWObjectStoreOperation<TObjectContext>(byte[] key, ArgSlice
             Debug.Assert(_output.spanByteAndMemory.IsSpanByte);
 
 
-            if (!status.Record.Created && !status.Record.CopyUpdated && !status.Record.InPlaceUpdated)
+            if (status.NotFound || (!status.Record.Created && !status.Record.CopyUpdated && !status.Record.InPlaceUpdated))
                 return GarnetStatus.NOTFOUND;
```

Comment:
We should not need CopyUpdated or InPlaceUpdated to be mentioned here; NotFound cannot be true if they are present. Look at RMW_MainStore (AdvancedOps.cs); it is much more straightforward.

           return status.Found || status.Record.Created ? GarnetStatus.OK : GarnetStatus.NOTFOUND;